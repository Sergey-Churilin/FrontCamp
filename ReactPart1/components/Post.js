import React from 'react';

class Post extends React.Component {
    render() {
        const post = this.props.post;

        return (
            <div className="post">
                {
                    <div>
                        <h3 className="author">{post.author}</h3>
                        <div>{post.post}</div>
                        <button className="delete" onClick={()=>{console.log(this.props);this.props.deletePost(post.id)}}>X</button>
                    </div>
                }
            </div>
        );
    }
}
export default Post;

