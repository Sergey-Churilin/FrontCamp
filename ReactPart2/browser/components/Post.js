import React from 'react';
import {mapActionsToProps, mapStateToProps} from "../selectors/Selectors";
import {connect} from "react-redux";

class Post extends React.Component {
    render() {
        const post = this.props.post;
        const visible = this.props.authorization.isLoggedIn;
        const className = visible ? "delete" : "delete invisible";
        return (
            <div className="post">
                {
                    <div>
                        <h3 className="author">{post.author}</h3>
                        <div>{post.post}</div>
                        <button className={className} onClick={()=>{this.props.deletePost(post._id,this.props.authorization.token)}}>X</button>
                    </div>
                }
            </div>
        );
    }
}

module.exports = Post;

