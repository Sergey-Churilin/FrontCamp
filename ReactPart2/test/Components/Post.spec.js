import React from 'react';
import Post from '../../browser/components/Post';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const post = {
        _id: 1,
        author: 'author',
        post: 'text'
    };

    const component = renderer.create(
        <Post key={post._id} post={post} onClick={() => {}} authorization={{isLoggedIn :true}}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});