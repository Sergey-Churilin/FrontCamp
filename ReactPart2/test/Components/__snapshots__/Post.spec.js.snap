// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="post"
>
  <div>
    <h3
      className="author"
    >
      author
    </h3>
    <div>
      text
    </div>
    <button
      className="delete"
      onClick={[Function]}
    >
      X
    </button>
  </div>
</div>
`;
